build.boot

```diff
@@ -1,80 +1,163 @@
+   [com.amazonaws/aws-java-sdk "1.11.6"]
 
-                            ;; HTTP Server
-                            [io.pedestal/pedestal.service "0.5.2"]
-                            [io.pedestal/pedestal.jetty   "0.5.2"]
-                            [io.pedestal/pedestal.interceptor "0.5.2"]
+   ;; HTTP Server
+   [io.pedestal/pedestal.service "0.5.2"]
+   [io.pedestal/pedestal.jetty   "0.5.2"]
+   [io.pedestal/pedestal.interceptor "0.5.2"]
 
-                            ;; HTTP Client
-                            [http-kit "2.3.0-alpha1"]
+   ;; HTTP Client
+   [http-kit "2.3.0-alpha1"]
 
-                            ;; URL util lib
-                            [com.cemerick/url "0.1.1"]
+   ;; URL util lib
+   [com.cemerick/url "0.1.1"]
 
-                            ;; Repl reloading
-                            [reloaded.repl "0.2.3" :scope "test"]
+   ;; Repl reloading
+   [reloaded.repl "0.2.3" :scope "test"]
 
-                            ;; Testing
-                            [adzerk/boot-test "1.1.2" :scope "test"]
+   ;; Testing
+   [adzerk/boot-test "1.1.2" :scope "test"]
 
-                            ;; Code Analysis
-                            [tolitius/boot-check "0.1.4" :scope "test"]]
+   ;; Code Analysis
+   [tolitius/boot-check "0.1.4" :scope "test"]]
+)
+(def datomic-free
+  '[[com.datomic/datomic-free "0.9.5554"]])
+
+(def datomic-pro
+  '[[com.datomic/datomic-pro "0.9.5554"]])
+
+(defn- get-wombats-env
+  "Gets the user defined wombats environment to determine dependencies"
+  []
+  (let [env (System/getenv "WOMBATS_ENV")]
+    (if (or (= env "")
+            (nil? env))
+      "dev"
+      env)))
+
+(defn- get-wombats-db-name
+  "Gets the user readable name of the database that various tasks will be performed on"
+  []
+  (let [wombats-env (get-wombats-env)]
+    (case wombats-env
+      "dev" "local"
+      "dev-ddb" "development"
+      "qa-ddb" "qa"
+      "prod-ddb" "production")))
+
+(defn- get-env-permissions
+  "Used by can-run-command to determine if various functions can be called on the specified db"
+  []
+  (let [wombats-env (get-wombats-env)]
+    (case wombats-env
+      "dev" ["refresh" "delete" "seed" "refresh-db-functions"]
+      "dev-ddb" ["refresh" "delete" "seed" "refresh-db-functions"]
+      "qa-ddb" ["refresh" "delete" "seed" "refresh-db-functions"]
+      "prod-ddb" [])))
+
+(defn- can-run-command?
+  "Uses environment to check whether a command string can be run"
+  [cmdstring]
+  (if (not= (some #(= cmdstring %) (get-env-permissions)) nil)
+    true
+    false)
+  )
+
+(defn- get-dependencies
+  "Checks environment to determine whether to load datomic free or pro"
+  []
+  (let [env (get-wombats-env)]
+    (if (= env "dev")
+      (into main-dependencies datomic-free)
+      (into main-dependencies datomic-pro)
+      )))
+
+(defn- is-dev?
+  "Check if current env is dev or dev-ddb for loading user"
+  []
+  (if (or (= (get-wombats-env) "dev") (= (get-wombats-env) "dev-ddb"))
```

Comment:
I would let this since you use the same variable twice. Also I'd move the second equality below so the `or` is more clear.